Tour steps functionality

bartaz opened the pull request:
Fixes #2198 

- Moves demo tour to publisher listing page
- Adds support for multiple steps with content provided via JSON
- Adds support for different positioning of tooltips to adjust to different elements on page (2 possible positions for the sake of demo)

### QA
This is not a complete feature. It's a minimal functional demo of the tour, merged to a feature branch where the work on it will continue.

- ./run or demo
-  sign in and go to listing page of any snap you own
- find a tour question mark (?) button in right bottom corner
- click it to trigger the tour
- navigate between the steps
- skip or finish tour to close
- tour should handle scrolling and resizing of the screen

<img width="1154" alt="Screenshot 2019-08-23 at 13 26 05" src="https://user-images.githubusercontent.com/83575/63589544-ba76bf00-c5a9-11e9-89d1-a4fa4ea2cfa5.png">

webteam-app commented:
Starting demo at: https://snapcraft-io-canonical-web-and-design-pr-2208.run.demo.haus/

therealjuan commented:
* When resizing there is a moment that the window gets funny so not sure if we want to hide completely the component while resizing or maybe it is due to the animation?

![image](https://user-images.githubusercontent.com/8167677/63601932-54486700-c5be-11e9-850c-449638436b18.png)

* Maybe we need a little bit more margin on the top and bottom of the mask overlay?
![image](https://user-images.githubusercontent.com/8167677/63601995-7510bc80-c5be-11e9-88c6-fe296d8feef1.png)

bartaz commented:
> When resizing there is a moment that the window gets funny so not sure if we want to hide completely the component while resizing or maybe it is due to the animation?

There is some delay before we recalculate position during resize (because we don't want to do that thousands of times when user is resizing). But the weird behaviour is probably more visible for these right positioned steps, because the tooltip in this case is positioned relative to window right, while the mask to the window left, so when you resize, tooltip stays close to window border while mask stays behind.
As suggested, the easiest workaround would be to hide either the mask 'window' or both the mask and tooltip step while resizing and show them in new position once resize is finished.

> Maybe we need a little bit more margin on the top and bottom of the mask overlay?

The mask shape relies a lot on spacing around targeted element (with a 5px offset). We can easily make the offset bigger. There will be cases where mask will look misaligned because top and bottom spacing have different values - it's a common case in Vanilla, and we can't do much about it.
But making the offset bigger will at least give elements more space in the highlight.

bartaz commented:
@therealjuan I updated the mask offset to make it a bit bigger, but want to keep it within small spacing of Vanilla to avoid mask showing parts of other elements around.

Also added some functionality to hide the mask when resizing, so it looks less weird.
